### DISCUSSION TITLE
Linters Configuration Discussion

### DISCUSSION THREAD

**Discussion by frankie-mur** [kmesiab/orion#29] [General]
In this discussion thread, we aim to explore and establish best practices for configuring linters in our project. Linters play a crucial role in maintaining code quality and consistency, and we need to align on a set of guidelines that benefit the entire development process.

Topics for Discussion:

- Linting Tools: Share your experiences with different linting tools and their suitability for our project.
- Configuration Standards: Discuss and propose standards for configuring linters to ensure consistency across the codebase.
- Custom Rules: Explore the possibility of implementing custom linting rules tailored to our project's requirements.
- Integration with CI/CD: Share insights on integrating linters into our continuous integration and deployment pipelines.

Feel free to share your thoughts, experiences, and suggestions regarding linters configuration. Let's collaborate to enhance our coding practices and maintain a high standard of code quality in the project! 🚀

**Comment by kmesiab** [comment]
**First, ensure you use the linters as this will ensure code style is followed.**

What IDE's are in use?  I use Goland for Go and VSCode for Typescript (unless I'm working on both at the same time, then I will work in Goland exclusively. 

I have:

1. **Line breaks**: Hard break at 85 characters
[Will keep updating this list given time]
2. 

Best Practices:

**Ensure each file has an endine line break.**  You can tell if you missed, GitHub will show a red circle at the bottom of any file in a PR that lacks one.


Goland Go Formatting Settings

![image](https://github.com/kmesiab/orion/assets/161768/32199765-458e-4ba6-ac41-71b0002cb838)

![image](https://github.com/kmesiab/orion/assets/161768/e69c8fcc-c18b-4e11-ac77-f1f8dab4b2c2)
